packages/filebrowser/src/listing.ts

```diff
@@ -1824,6 +1905,16 @@ export namespace DirListing {
      */
     getNameNode(node: HTMLElement): HTMLElement;
 
+    /**
+     * Get the checkbox input element node.
+     *
+     * @param node A node created by [[createItemNode]] or
+     * [[createHeaderItemNode]]
+     *
+     * @returns The checkbox node.
+     */
+    getCheckboxNode?: (node: HTMLElement) => HTMLInputElement;
```

Comment:
I wasn't sure if I should modify this interface (`DirListing.IRenderer`) in a backwards-incompatible way, so I made this method optional. I was also careful in the `DirListing` methods to not assume that the instance's renderer comes with checkboxes, which is why you see a lot of the code wrapped in `if (renderer.getCheckboxNode)`.